tests/integration-tests/tests/performance_tests/test_startup_time.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import statistics
+from datetime import datetime
+
+import boto3
+import pytest
+from dateutil.relativedelta import relativedelta
+from utils import describe_cluster_instances
+
+MINIMUM_DATASET_SIZE = 2
+
+
+def evaluate_data(value, data):
+    standard_deviation = statistics.stdev(data)
+    mean = statistics.mean(data)
+    logging.info(f"Standard deviation: {standard_deviation}")
+
+    if value < (mean + 2 * standard_deviation) or value > (mean - 2 * standard_deviation):
+        return False
+    return True
+
+
+def get_data(dimensions, cw_client):
+    data = []
+
+    cluster_metrics = cw_client.list_metrics(
+        Namespace="ParallelCluster", MetricName="StartupTime", Dimensions=dimensions
+    )
+
+    logging.info(f"Metrics: {cluster_metrics}")
+
+    for metric in cluster_metrics["Metrics"]:
+        result = cw_client.get_metric_statistics(
+            Namespace="ParallelCluster",
+            MetricName="StartupTime",
+            Dimensions=metric["Dimensions"],
+            StartTime=datetime.now() - relativedelta(years=1),
+            EndTime=datetime.now(),
+            Period=30000,
+            Statistics=["Average"],
+            Unit="None",
+        )
+
+        logging.info(f"Results: {result}")
+
+        if result["Datapoints"]:
+            value = result["Datapoints"][0].get("Average")
+            data.append(value)
+    return data
+
+
+def get_metric(dimensions, cw_client):
+    value = None
+
+    result = cw_client.get_metric_statistics(
+        Namespace="ParallelCluster",
+        MetricName="StartupTime",
+        Dimensions=dimensions,
+        StartTime=datetime.now() - relativedelta(years=1),
+        EndTime=datetime.now(),
+        Period=30000,
+        Statistics=["Average"],
+        Unit="None",
+    )
+
+    logging.info(f"Results: {result}")
+
+    if result["Datapoints"]:
+        value = result["Datapoints"][0].get("Average")
+    return value
+
+
+def test_startup_time(pcluster_config_reader, clusters_factory, test_datadir, region, instance, os, scheduler):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    logging.info("Cluster Created")
+
+    cw_client = boto3.client("cloudwatch", region_name=region)
+
+    instances = describe_cluster_instances(
+        cluster.name,
+        region,
+        filter_by_node_type="Compute",
+    )
+
+    performance_degradation = {}
+
+    for instance in instances:
+        instance_type = instance["InstanceType"]
+        instance_id = instance["InstanceId"]
+        logging.info(f"Type: {instance_type}")
+
+        # get new startup time
+
+        dimensions = [
+            {"Name": "OS", "Value": os},
+            {"Name": "InstanceID", "Value": instance_id},
```

Comment:
Specifying the instance id should be enough to retrieve the metric emitted by an ec2 instance.
why do we need to specify also the OS, cluster name and instance_type?

Reply:
In order for the get_metric_statistics call to return the data point, you have to include all the dimensions. I tried only specifying the instance id, but the api call returned an empty dataset.

Reply:
Got it -- Having a look at the design I also understand the reasoning behind it. Let's go with this implementation.